src/Raven.Server/Rachis/Candidate.cs

```diff
@@ -97,6 +97,10 @@ private void Run()
                         };
                         _engine.AppendStateDisposable(this, candidateAmbassador); // concurrency exception here will dispose the current candidate and it ambassadors
                         candidateAmbassador.Start();
+                    }
+                    foreach (var witness in clusterTopology.Witnesses)
```

Comment:
You can merge this with the behavior for `Members` above, exact same thing needs to be done here.